tests/autopilot-capacity/aut_test.go

```diff
@@ -41,63 +102,118 @@ var _ = Describe(fmt.Sprintf("{%sBasic}", testNameSuite), func() {
-					StorageProvisioner:  Inst().Provisioner,
-					AutopilotParameters: apParameters,
-				})
-				Expect(err).NotTo(HaveOccurred())
-				Expect(context).NotTo(BeEmpty())
-				contexts = append(contexts, context...)
+		for _, apRule := range AutopilotruleBasicTestCases {
+			apParameters := &scheduler.AutopilotParameters{
+				Enabled:                 true,
+				Name:                    testName,
+				AutopilotRuleParameters: apRule,
 			}
-		})
 
-		Step("wait until workload completes on volume", func() {
-			for _, ctx := range contexts {
-				err = Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
-				Expect(err).NotTo(HaveOccurred())
-			}
-		})
-
-		Step("validating volumes and verifying size of volumes", func() {
-			for _, ctx := range contexts {
-				err = Inst().S.InspectVolumes(ctx, timeout, retryInterval)
-				Expect(err).NotTo(HaveOccurred())
-			}
-		})
+			Step("schedule applications", func() {
+				for i := 0; i < Inst().ScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%v", fmt.Sprintf("%s-%d", testName, i), Inst().InstanceID)
+					context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+						AppKeys:             Inst().AppList,
+						StorageProvisioner:  Inst().Provisioner,
+						AutopilotParameters: apParameters,
+					})
+					Expect(err).NotTo(HaveOccurred())
+					Expect(context).NotTo(BeEmpty())
+					contexts = append(contexts, context...)
+				}
+			})
+
+			Step("wait until workload completes on volume", func() {
+				for _, ctx := range contexts {
+					err = Inst().S.WaitForRunning(ctx, workloadTimeout, retryInterval)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			Step("validating volumes and verifying size of volumes", func() {
+				for _, ctx := range contexts {
+					err = Inst().S.InspectVolumes(ctx, timeout, retryInterval)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			Step("wait for unscheduled resize of volume (10min)", func() {
+				time.Sleep(10 * time.Minute)
+			})
+
+			Step("validating volumes and verifying size of volumes", func() {
+				for _, ctx := range contexts {
+					err = Inst().S.InspectVolumes(ctx, timeout, retryInterval)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			Step("destroy apps", func() {
+				opts := make(map[string]bool)
+				opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+				for _, ctx := range contexts {
+					TearDownContext(ctx, opts)
+				}
+			})
+		}
+	})
+})
 
-		Step("wait for unscheduled resize of volume (10min)", func() {
-			time.Sleep(10 * time.Minute)
-		})
+// This test id using for scale testing of Autopilot. It resizes the volume
+// untill the size of the volume will be more than desired
+var _ = Describe(fmt.Sprintf("{%sScale}", testNameSuite), func() {
+	It("will resize the volume until the max size of the volume", func() {
+		var contexts []*scheduler.Context
+		var err error
+		testName := strings.ToLower(fmt.Sprintf("%sScale", testNameSuite))
```

Comment:
On second thoughts, `Scale` might be a misnomer here. Rename this to Parallel since this is what we are trying to do. i.e start many pvc resizes in parallel.

The scale should be achieved by using `ScaleFactor`.

Reply:
You need to rename `Scale` to `DoesNotWaitForWorkload`